lib/shopping_cart.rb

```diff
@@ -0,0 +1,32 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def capacity
```

Comment:
The problem with this method is ruby never sees it, since we have an `attr_reader :capacity` on line 1, that method in the attr_reader will override whatever method definition we have here.